libcst/codemod/visitors/_apply_type_annotations.py

```diff
@@ -0,0 +1,558 @@
+# Copyright (c) 2016-present, Facebook, Inc.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree
+
+
+from typing import Dict, List, NamedTuple, Optional, Sequence, Set, Tuple, Union
+
+import libcst as cst
+from libcst.codemod._context import CodemodContext
+from libcst.codemod._visitor import ContextAwareTransformer
+from libcst.codemod.visitors._gather_imports import GatherImportsVisitor
+
+
+def _get_name_or_attribute_as_string(
+    name_or_attribute: cst.BaseExpression,
```

Comment:
Can you give this a more concrete Union type? As I'm going through the ones below it looks like this might be `Union[cst.Name, cst.Attribute, None]` and anything else is intended as an error?

Reply:
It's passed a `BaseExpression` in a few places. For example, `Attribute.value` gives a `BaseExpression`. Given that it's a tradeoff between doing `isinstance` in the caller or returning `Optional` in this function, I feel it's better to return `Optional` since it returns `Optional` for some Attributes anyway.

Reply:
It's a very common pattern to get the dot-concatenated name given a Name, Attribute or other type of nodes, and we already built a helper for it.
https://libcst.readthedocs.io/en/latest/helpers.html#libcst.helpers.get_full_name_for_node
It returns exactly `Optional[str]`. Please use this one.